client/containers/DashboardPages/CreatePageDialog.tsx

```diff
@@ -78,8 +79,18 @@ class CreatePageDialog extends Component<Props, State> {
 				window.location.href = getDashUrl({ mode: 'pages', subMode: newPageResult.slug });
 			})
 			.catch((err) => {
-				console.error(err);
-				this.setState({ isLoading: false, error: err });
+				const slugStatus: SlugStatus =
+					err?.type === 'forbidden-slug' ? err.slugStatus : 'available';
+				if (slugStatus !== 'available') {
+					// eslint-disable-next-line react/no-access-state-in-setstate
```

Comment:
what does "is deferred and batched by React" mean?

Reply:
Pulling this from the [documentation for this lint rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-access-state-in-setstate.md):

```ts
this.setState({value: this.state.value + 1}) // 2
this.setState({value: this.state.value + 1}) // 2, not 3
```

This is a silly example of code no one would ever write, but it becomes more realistic if you consider wrapping one of those lines in an `increment()` function which is called several times in a long and gnarly piece of code.

When `setState()` is called, it doesn't update the value in `this.state` immediately, but it batches all of the requested state updates together and executes them all at once. So if you want later `setState()` calls to operate on the values that the earlier ones produced, it's best to pass a function into `setState()`. In this case it doesn't matter because there are no previous calls to `setState()` — so in a sense we know better than the lint rule here, but it's harmless to follow it. Off the top of my head I don't know whether a lint rule _could_ be written to only catch the actually problematic cases here, but in any case they haven't done so.

FYI: there is [even more powerful batching](https://github.com/reactwg/react-18/discussions/21) coming to React 18!